packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -1281,6 +1283,43 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     }
   }
 
+  /**
+   * Retrieve feature flag information from the cloud assembly
+   */
+  public async flags(cx: ICloudAssemblySource): Promise<FeatureFlagReportProperties[]> {
+    const ioHelper = asIoHelper(this.ioHost, 'flags');
+    const assembly = await assemblyFromSource(ioHelper, cx);
+    const artifacts = assembly.cloudAssembly.manifest.artifacts;
+    const allFeatureFlags: ArtifactManifest[] = [];
+
+    for (const [_, artifactInfo] of Object.entries(artifacts!)) {
```

Comment:
This works, but here's  a way to do this in fewer lines:

```ts
const allFeatureFlags = Object.values(artifacts).filter(a => a.type === ArtifactType.FEATURE_FLAG_REPORT);
```

The pattern "take a collection, iterate over it, and accumulate some entries into another collection" is very common and can be expressed more directly (imo, at least).